R/conjugate_encoder.R

```diff
@@ -31,6 +36,159 @@
+
+      }
+
+      train_processed <- reduce(encodings_list, function(x, y) left_join(x, y, by = NULL)) %>%
+        select(-all_of(cat_columns))
+
+      encodings_list[[1]] <- X_test
+
+      test_encodings_means <- reduce(encodings_list, function(x, y) left_join(x, y, by = NULL)) %>%
+        select(contains("encoded_mean"))
+
+      test_encodings_means[is.na(test_encodings_means)] <- mu
+
+      test_encodings_var <- reduce(encodings_list, function(x, y) left_join(x, y, by = NULL)) %>%
+        select(contains("encoded_var"))
+
+      test_encodings_var[is.na(test_encodings_var)] <- beta / (alpha - 1)
+
+      test_processed <- bind_cols(X_test, test_encodings_means) %>%
+        bind_cols(., test_encodings_var) %>%
+        select(-all_of(cat_columns))
+
+
+    } else {
+
+      if (length(unique(y)) != 2) {
+        stop("Binary classification is supported only.")
+      }
+
+      if (setequal(c("alpha", "beta"), names(prior_params)) == FALSE) {
+        stop("Missing a required prior parameter.")
+      }
+
+      if (prior_params$alpha <= 0 | prior_params$beta <= 0) {
+        stop("Invalid prior specification. Note that alpha and beta must all be greater than 0.")
+      }
+
+      if (is.factor(y) || is.character(y)) {
+
+        all_unique_values <- unique(y)
+        target_new <- case_when(y == all_unique_values[1] ~ 0,
+                                y == all_unique_values[2] ~ 1,
+                                TRUE ~ y)
+
+      }
+
+      encodings_list <- vector("list", length(cat_columns) + 1)
+      encodings_list[[1]] <- X_train
+      alpha <- prior_params$alpha
+      beta <- prior_params$beta
+      n <- nrow(X_train)
+
+      for (i in seq_along(cat_columns)) {
+
+        column <- cat_columns[i]
+
+        conditionals <- X_train %>%
+          bind_cols(target = y) %>%
+          group_by(!!sym(column)) %>%
+          summarize(conditional_success = sum(target))
+
+        alpha_post <- alpha + conditionals$conditional_success
+        beta_post <- beta + n - conditionals$conditional_success
+
+        posterior_expected_val <- alpha_post / (alpha_post + beta_post)
+        all_encodings <- conditionals %>%
+          bind_cols(., tibble(encoded = posterior_expected_val)) %>%
+          select(-conditional_success)
+
+        names(all_encodings)[2] <- paste(column, "encoded", sep = "_")
+
+        encodings_list[[i + 1]] <- all_encodings
+
+      }
+
+      train_processed <- reduce(encodings_list, function(x, y) left_join(x, y, by = NULL)) %>%
+        select(-all_of(cat_columns))
+
+      encodings_list[[1]] <- X_test
+
+      test_encodings <- reduce(encodings_list, function(x, y) left_join(x, y, by = NULL)) %>%
+        select(contains("encoded"))
+
+      test_encodings[is.na(test_encodings)] <- alpha / (alpha + beta)
+
+      test_processed <- bind_cols(X_test, test_encodings) %>%
+        select(-all_of(cat_columns))
+
+    }
+
   out <- list(train_processed, test_processed)
```

Comment:
I have this list with named values so maybe for consistency this should be changed 
```suggestion
  out <- list("train" = train_processed, "test" = test_processed)
```

Reply:
Sure, sounds good.